### PR TITLE
adding security_policy field in regionBackendService resource

### PR DESCRIPTION
**Pull request by felipegc**
fixes:
https://github.com/hashicorp/terraform-provider-google/issues/14987

Adding security_policy field in google_compute_region_backend_service resource.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Compute: added `security_policy` field to `google_compute_region_backend_service` resource(beta)
```

### CODE REVIEW

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like there are merge conflicts. Can you fix those?

**Comment by felipegc** [comment]
@slevenick Thanks for warning me that. I fixed the conflicts. Could you please continue with the code review?

**Comment by felipegc** [comment]
@slevenick Is there anything else I have to do to proceed with the code review? Thanks

File: mmv1/templates/terraform/post_create/compute_region_backend_service_security_policy.go.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  region, err := tpgresource.GetRegion(d, config)
+  if err != nil {
+    return err
+  }
+
+  spr := emptySecurityPolicyReference()
+  spr.SecurityPolicy = pol.RelativeLink()
+
+  op, err := config.NewComputeClient(userAgent).RegionBackendServices.SetSecurityPolicy(project, region, obj["name"].(string), spr).Do()
```
Comment (slevenick) [inline mmv1/templates/terraform/post_create/compute_region_backend_service_security_policy.go.erb:17]: This resource works via the generated HTTP calls, I think that introducing a requirement on the client library here is going to be an issue. Can we make this call directly to the API? Maybe using some of the more complex magic-modules functionality like update_url and update_verb on the field itself? See https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/compute/Address.yaml#L201 as an example

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/post_create/compute_region_backend_service_security_policy.go.erb (hunk 1 above)
Comment (felipegc) [reply mmv1/templates/terraform/post_create/compute_region_backend_service_security_policy.go.erb]: Done.

**Comment by felipegc** [comment]
@slevenick Is there anything else from my side? Thanks

**Review by slevenick — APPROVED** [review]

**Comment by felipegc** [comment]
> Looks good to me. Only issue is that the YAML file is not passing linting, can you fix that?
> 
> I'm not sure exactly why, it looks fine to my eye, but you may need to run the linter locally to see the error

Thanks for pointing that out. I had duplicated the "min_version: beta" definition. Ready for review again.

**Review by slevenick — APPROVED** [review]

**Pull request merged**